### PR TITLE
Fix: Missing syntax highlighting for css partials

### PR DESCRIPTION
**Pull request by rizwan3d**
# Pull Request

## 📖 Description

Missing syntax highlighting for CSS partials in FAST tagged templates extension.

### 🎫 Issues

Regex only has CSS for consideration for template. 
fix: Missing syntax highlighting for css partials in FAST tagged templates extension #6380

### General

- [ ] I have included a change request file using `$ yarn change`
- [X] I have added tests for my changes.
- [X] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [ ] I have read the [CONTRIBUTING](https://github.com/microsoft/fast/blob/master/CONTRIBUTING.md) documentation and followed the [standards](/docs/community/code-of-conduct/#our-standards) for this project.

### Component-specific

- [ ] I have added a new component
- [X] I have modified an existing component
- [ ] I have updated the [definition file](https://github.com/microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#definition)
- [ ] I have updated the [configuration file](https://github.com/microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#configuration)

### CODE REVIEW

**Comment by rizwan3d** [comment]
@microsoft-github-policy-service agree [company="{GrowBit Tech}"]

File: packages/tooling/fast-vscode-syntax-highlighting/syntaxes/fast-tagged-templates.tmLanguage.json [hunk 1]
```diff
@@ -15,7 +15,7 @@
     ],
     "repository": {
         "css-tagged-template": {
-            "begin": "(?i)(?<!\\.)(css|((?:\\/\\*\\s*?)css(?:\\s*\\*\\/)))\\s*(`)",
+            "begin": "(?i)(?<!\\.)((cssPartial|css)|((?:\\/\\*\\s*?)(cssPartial|css)(?:\\s*\\*\\/)))\\s*(`)",
```
Comment (KingOfTac) [inline packages/tooling/fast-vscode-syntax-highlighting/syntaxes/fast-tagged-templates.tmLanguage.json:18]: The syntax for this has changed for the in development versions of FAST so this should be updated to handle that as well. `cssPartial` was deprecated and added as an extension to `css` in the form of `css.partial`

**Review by KingOfTac — CHANGES_REQUESTED** [review]

**Comment by rizwan3d** [comment]
@microsoft-github-policy-service agree

File: packages/tooling/fast-vscode-syntax-highlighting/syntaxes/fast-tagged-templates.tmLanguage.json [hunk 2]
```diff
@@ -15,7 +15,7 @@
     ],
     "repository": {
         "css-tagged-template": {
-            "begin": "(?i)(?<!\\.)(css|((?:\\/\\*\\s*?)css(?:\\s*\\*\\/)))\\s*(`)",
+            "begin": "(?i)(?<!\\.)((css(\\.)?(partial)?)|((?:\\/\\*\\s*?)(css(\\.)?(partial)?)(?:\\s*\\*\\/)))\\s*(`)",
```
Comment (radium-v) [inline packages/tooling/fast-vscode-syntax-highlighting/syntaxes/fast-tagged-templates.tmLanguage.json:18]: Adding capture groups (parentheses) changes the order of the groups used in the `beginCaptures` property. These changes add six new capture groups, which would push the `punctuation.definition.string.template.begin.ts` (which handles the backtick that starts the template literal) from capture 3 to capture 9.

Comment (radium-v) [reply packages/tooling/fast-vscode-syntax-highlighting/syntaxes/fast-tagged-templates.tmLanguage.json:18]: Modifying the capturing groups to be non-blocking can get it down to four. The `.` in `css.partial` needs to be captured in the first half to treat it as a property accessor, and I added a `\b` to ensure that we're only matching `css.partial` and not something like `SomeCoolLibraryForCSS.partial`.

```ts
            "begin": "(?i)(?<!\\.)(\\bcss(?:(\\.?)partial)|((?:\\/\\*\\s*?)css(?:\\.?partial)?(?:\\s*\\*\\/)))\\s*(`)",
            "beginCaptures": {
                "1": {
                    "name": "entity.name.function.tagged-template.ts"
                },
                "2": {
                    "name": "punctuation.accessor.ts"
                },
                "3": {
                    "name": "comment.block"
                },
                "4": {
                    "name": "punctuation.definition.string.template.begin.ts"
                }
            },
```

**Review by radium-v — CHANGES_REQUESTED** [review]

**Comment by radium-v** [comment]
I think since `css.partial` allows for just property lists (instead of selectors followed by property lists), this may need some more work to function properly. There aren't many uses of `css.partial` in the monorepo but here's one from a test:

```ts
const partial = css.partial`color: ${new myDirective}`;
```

This pattern would need to be defined on its own so it can include the weirdly named `source.css#rule-list-innards` pattern.

**Review by KingOfTac — APPROVED** [review]

**Comment by radium-v** [comment]
This needs to be held since the `vscode-css` package can either handle key-value pairs, or selectors, but not both at the same time.

microsoft/vscode-css#9

**Review by radium-v — CHANGES_REQUESTED** [review]
Holding for further investigation

**Comment by janechu** [comment]
Closing this PR as stale and obsolete with the migration of the package to @radium-v, see #6964.

**Pull request closed without merging**